File: Python/Algorithm/baekjoon/10828_스택.py
# baekjoon source = "https://www.acmicpc.net/problem/10828"
import sys

T = int(sys.stdin.readline())

stack = [0]*T
tmp = -1
for test_case in range(T):
    
    N = sys.stdin.readline().split()

    if len(N) > 1:
        cmd = N[0]
        number = N[1]
    else:
        cmd = N[0]
    
    if cmd == 'push':
        tmp += 1
        stack[tmp] = number
    elif cmd == 'pop':
        if tmp == -1:
            print(-1)
        else:
            print(stack[tmp])
            stack[tmp] = 0
            tmp -= 1
    elif cmd == 'size':
        print(tmp+1)
    elif cmd == 'empty':
        if tmp == -1:
            print(1)
        else:
            print(0)
    elif cmd == 'top':
        if tmp == -1:
            print(-1)
        else:
            print(stack[tmp])

File: Java/baekjoon/화목한스터디/src/스택/_10828_스택.java
//baekjoon source = "https://www.acmicpc.net/problem/10828"
package 스택;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.StringTokenizer;

public class _10828_스택 {
	static int[] arr = new int[10001];
	static int top = -1;
	static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
	static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

	static void push(int x) {
		arr[++top] = x;
	}

	static void pop() throws IOException {
		if (top == -1) {
			bw.write(-1 + "\n");
		} else {
			bw.write(arr[top] + "\n");
			top--;
		}
	}

	static void size() throws IOException {
		bw.write(top + 1 + "\n");
	}

	static void empty() throws IOException {
		bw.write((top == -1 ? 1 : 0) + "\n");
	}

	static void top() throws IOException {
		bw.write((top == -1 ? -1 : arr[top]) + "\n");
	}

	public static void main(String[] args) throws IOException {
		int n = Integer.parseInt(br.readLine());
		for (int i = 0; i < n; i++) {
			StringTokenizer st = new StringTokenizer(br.readLine());
			switch (st.nextToken()) {
			case "push":
				push(Integer.parseInt(st.nextToken()));
				break;
			case "pop":
				pop();
				break;
			case "size":
				size();
				break;
			case "empty":
				empty();
				break;
			default:
				top();
			}
		}
		bw.flush();
	}
}
